File: 100-Days-Series/Day 46/Stack.js
class StackNode{
    constructor(data){
        this.data=data;
        this.next=null;
    }
}
class Stack{
    constructor(){
        this.top=null;
    }
    push(data){
        const   newNode=new StackNode(data);
        newNode.next=this.top;
        this.top=newNode;
        console.log("New Data Added");
    }
    pop(){
        if(this.top==null){
            console.log("Stack is empty!");
        }else{
            console.log("Popped element is : "+this.top.data);
            this.top=this.top.next;
        }
    }
    peek(){
        if(this.top==null){
            console.log("Stack is empty!");
        }else{
            console.log("Top element is : "+this.top.data);
        }
    }
    traverse(){
        if(this.top==null){
            console.log("Stack is empty!");
        }else{
            let current=this.top;
            while(current!=null){
                console.log("Data is : "+current.data);
                console.log("--------------------------");
                current=current.next;
            }
        }
    }
    isEmpty() {
        console.log(this.top ? "Stack is not Empty" : "Stack is Empty");
    }
}
// Let's Start here
const stack =new Stack();
stack.push(10);
stack.push(20);
stack.push(30);
stack.push(40);
stack.traverse();
stack.peek();
stack.pop();
stack.traverse();
stack.isEmpty();

File: 100-Days-Series/Day 46/Stack.py
class StackNode:
    def __init__(self,data):
        self.data=data
        self.next=None
    
class Stack:
    def __init__(self):
        self.top=None


    def push(self,data):
        newNode=StackNode(data)
        newNode.next=self.top
        self.top=newNode
        print('New Data Added')

    def pop(self):
        if not self.top:
            print('Stack is empty!')
        else:
            print('Popped element is ',self.top.data)
            self.top=self.top.next
    
    def peek(self):
        if not self.top:
            print('Stack is empty!')
        else:
            print('Top element is ',self.top.data)

    def traverse(self):
        if not self.top:
            print('Stack is empty!')
        else:
            current=self.top
            while current:
                print('Data is : ',current.data)
                current=current.next

    def is_empty(self):
        if not self.top:
            print('Stack is empty')
        else:
            print('Stack is not empty')

stack=Stack()
stack.push(10)
stack.push(20)
stack.push(30)
stack.push(40)
stack.traverse()
stack.peek()
stack.pop()
stack.traverse()
stack.is_empty()